dataflow: Update taint-mode docs with field sensitivity (#773)
returntocorp/semgrep-docs: docs/writing-rules/data-flow/data-flow-overview.md

Semgrep strives for simplicity and delivers a lightweight, and fast static analysis. In addition to being intra-procedural, here are some other trade-offs:

- No path sensitivity: All _potential_ execution paths are considered, despite that some may not be feasible.

> - No pointer or shape analysis: _Aliasing_ that happens in non-trivial ways may not be detected, such as through arrays or pointers. Individual elements in arrays or other data structures are not tracked. There is limited field-sensitivity support in the taint engine, but none for constant propagation.

Reviewer: ```suggestion
- No pointer or shape analysis: _Aliasing_ that happens in non-trivial ways may not be detected, such as through arrays or pointers. Individual elements in arrays or other data structures are not tracked. The taint engine supports limited field sensitivity, but not yet for constant propagation.
```
